gusto/equations.py

```diff
@@ -879,6 +881,182 @@ def __init__(self, domain, parameters, fexpr=None, bexpr=None,
+    ∂D/∂t + H*∂u/∂x = 0,                                                     \n
+    for mean depth 'H', Coriolis parameter 'f' and gravity 'g'.
+
+    This is set up the from the underlying :class:`ShallowWaterEquations_1d`,
+    which is then linearised.
+    """
+
+    def __init__(self, domain, parameters, fexpr=None,
+                 space_names=None, linearisation_map='default',
+                 no_normal_flow_bc_ids=None, active_tracers=None):
+        """
+        Args:
+            domain (:class:`Domain`): the model's domain object, containing the
+                mesh and the compatible function spaces.
+            parameters (:class:`Configuration`, optional): an object containing
+                the model's physical parameters.
+            fexpr (:class:`ufl.Expr`, optional): an expression for the Coroilis
+                parameter. Defaults to None.
+            space_names (dict, optional): a dictionary of strings for names of
+                the function spaces to use for the spatial discretisation. The
+                keys are the names of the prognostic variables. Defaults to None
+                in which case the spaces are taken from the de Rham complex. Any
+                buoyancy variable is taken by default to lie in the L2 space.
+            linearisation_map (func, optional): a function specifying which
+                terms in the equation set to linearise. If None is specified
+                then no terms are linearised. Defaults to the string 'default',
+                in which case the linearisation includes both time derivatives,
+                the 'D' transport term, pressure gradient and Coriolis terms.
+            no_normal_flow_bc_ids (list, optional): a list of IDs of domain
+                boundaries at which no normal flow will be enforced. Defaults to
+                None.
+            active_tracers (list, optional): a list of `ActiveTracer` objects
+                that encode the metadata for any active tracers to be included
+                in the equations. Defaults to None.
+        """
+
+        if linearisation_map == 'default':
+            # Default linearisation is time derivatives, pressure gradient,
+            # Coriolis and transport term from depth equation
+            linearisation_map = lambda t: \
+                (any(t.has_label(time_derivative, pressure_gradient, coriolis))
+                 or (t.get(prognostic) == 'D' and t.has_label(transport)))
+
+        super().__init__(domain, parameters,
+                         fexpr=fexpr, space_names=space_names,
+                         linearisation_map=linearisation_map,
+                         no_normal_flow_bc_ids=no_normal_flow_bc_ids,
+                         active_tracers=active_tracers)
+
+        # Use the underlying routine to do a first linearisation of the equations
+        self.linearise_equation_set()
+        for t in self.residual:
```

Comment:
This print statement will need removing!

Reply:
urgh, sorry!